packages/tools/devtools/devtools-core/src/data-visualization/SharedTreeVisualizer.ts

```diff
@@ -0,0 +1,349 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { assert } from "@fluidframework/core-utils";
+import type {
+	FieldMapObject,
+	JsonableTree,
+	SharedTreeContentSnapshot,
+	TreeFieldStoredSchema,
+	TreeNodeStoredSchema,
+} from "@fluidframework/tree/internal";
+import {
+	EmptyKey,
+	LeafNodeStoredSchema,
+	MapNodeStoredSchema,
+	ObjectNodeStoredSchema,
+} from "@fluidframework/tree/internal";
+import type { SharedTreeLeafNode, VisualSharedTreeNode } from "./VisualSharedTreeTypes.js";
+import { type VisualChildNode, VisualNodeKind, type VisualValueNode } from "./VisualTree.js";
+
+/**
+ * Converts the output of {@link sharedTreeVisualizer} to {@link VisualChildNode} type containing `schema` and `children` fields.
+ */
+export function mapToVisualChildNode(tree: VisualSharedTreeNode): VisualChildNode {
+	if ("value" in tree) {
+		const result: VisualValueNode = {
+			value: tree.value,
+			nodeKind: VisualNodeKind.ValueNode,
+			sharedTreeSchemaData: tree.schema.allowedTypes,
+		};
+		return result;
+	} else {
+		const children: Record<string, VisualChildNode> = {};
+
+		for (const [key, value] of Object.entries(tree.fields)) {
+			const child = mapToVisualChildNode(value);
+			children[key] = child;
+		}
+
+		return {
+			children,
+			nodeKind: VisualNodeKind.TreeNode,
+			sharedTreeSchemaData: tree.schema.allowedTypes,
+		};
+	}
+}
+
+/**
+ * Returns the allowed fields & types for the object fields.
```

Comment:
This is a good start but for folks new to this code, it'd help for the comments to also describe what the params/return values mean.

Reply:
This applies to all the functions in this file